newIDE/app/src/UI/Carousel.js

```diff
@@ -0,0 +1,329 @@
+    },
+  },
+});
+
+const useStylesForGridListItem = makeStyles({
+  root: {
+    width: 'unset !important',
+  },
+  tile: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+});
+
+const Carousel = <ThumbnailType: Thumbnail>({
+  title,
+  items,
+  displayItemTitles = true,
+  tabIndexOffset = 0,
+}: Props<ThumbnailType>) => {
+  const theme = React.useContext(GDevelopThemeContext);
+  const [
+    shouldDisplayLeftArrow,
+    setShouldDisplayLeftArrow,
+  ] = React.useState<boolean>(false);
+  const classesForGridList = useStylesForGridList();
+  const classesForGridListItem = useStylesForGridListItem();
+  const scrollView = React.useRef<?HTMLUListElement>(null);
+  const itemsToDisplay =
+    items ||
+    Array(3)
+      .fill({
+        skeleton: true,
+        title: '',
+        description: '',
+        thumbnail: '',
+      })
+      .map((item, index) => ({ ...item, id: `skeleton${index}` }));
+
+  const windowSize = useResponsiveWindowWidth();
+  const imageHeight = referenceSizesByWindowSize.imageHeight[windowSize];
+  const arrowWidth = referenceSizesByWindowSize.arrowWidth[windowSize];
+  const cellWidth = (16 / 9) * imageHeight;
+  const widthUnit = cellWidth + cellSpacing;
+
+  const cellHeight =
+    imageHeight + (displayItemTitles ? titleHeight + spacerSize : 0);
+
+  const renderImage = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): React.Node => (
+      <img
+        src={item.thumbnailUrl}
+        style={{
+          ...styles.image,
+          height: imageHeight,
+          minHeight: imageHeight,
+          width: cellWidth,
+        }}
+        alt={item.title}
+        title={item.title}
+      />
+    ),
+    [cellWidth, imageHeight]
+  );
+
+  const renderThumbnail = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): ?React.Node => {
+      if (!item.skeleton && !item.link && !item.thumbnailUrl) return null;
+      if (item.link) {
+        return (
+          <a href={item.link} target="_blank">
+            {renderImage(item)}
+          </a>
+        );
+      }
+      if (item.thumbnailUrl) {
+        return renderImage(item);
+      }
+      if (item.skeleton) {
+        return (
+          <Skeleton variant="rect" height={imageHeight} width={cellWidth} />
+        );
+      }
+    },
+    [renderImage, cellWidth, imageHeight]
+  );
+
+  const renderItemTitle = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): ?React.Node => {
+      if (!displayItemTitles) return null;
+      return (
+        <>
+          <Spacer />
+          {item.title ? (
+            <div style={{ ...styles.itemTitleContainer, width: cellWidth }}>
+              <Text noMargin style={styles.itemTitle}>
+                {item.title}
+              </Text>
+            </div>
+          ) : (
+            <Skeleton
+              variant="rect"
+              height={titleHeight}
+              width={(cellWidth / 3) * (1 + 2 * Math.random())}
```

Comment:
I don't understand why there's a `Math.random()` at first reading. Maybe leave a comment?

Reply:
So that it feels more "alive". Without it, it would be rectangles of the same length
![image](https://user-images.githubusercontent.com/32449369/144078513-3e38b6b1-68bc-405f-a6e4-2fb0ade06099.png)

Reply:
Comment added